scripts/normalize-by-median.py

```diff
@@ -102,15 +109,23 @@ def __call__(self, input_filename, force_paired=False):
         desired_coverage = self.desired_coverage
         ksize = self.htable.ksize()
 
-        batch_size = 1
-        if force_paired:
-            batch_size = 2
+        screed_iter = screed.open(input_filename, parse_description=False)
+        reader = broken_paired_reader(screed_iter,
+                                      force_single=self.force_single,
+                                      require_paired=force_paired)
 
-        for batch in WithDiagnostics(input_filename, batch_size,
-                                     self.report_fp, force_paired, self):
+        for index, is_paired, read0, read1 in WithDiagnostics(input_filename,
+                                                              self.report_fp,
+                                                              force_paired,
+                                                              self, reader):
             passed_filter = False
             passed_length = True
 
+            batch = []
+            batch.append(read0)
+            if read1 is not None:
+                batch.append(read1)
+
             for record in batch:
                 if len(record.sequence) < ksize:
```

Comment:
note that broken_paired_reader can take a minimum sequence length (ksize), making this length check unused.  Although this may break the pairing code in this case... hmm...